### PR TITLE
Resolve Unit Test Build Warnings

### PR DESCRIPTION
**Pull request by HTRamsey**
Resolve Unit Test Build Warnings

Description
-----------
If you look at the build output, there are a ton of warnings in the unit tests. This resolves most of them. Also found some tests that were set up incorrectly.

Test Steps
-----------

Checklist:
----------

- [x] I have tested my changes. No regression in existing tests.
- [x] I have modified and/or added unit-tests to cover the code changes in this Pull Request.

Related Issue
-----------

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by tony-josi-aws** [comment]
@HTRamsey, thanks for this PR.

The endpoint structure change suggested in the PR could break existing projects:

```
/** @brief The description of an end-point. */
    typedef struct xNetworkEndPoint
    {
        union
        {
            struct
            {
                IPV4Parameters_t ipv4_settings; /**< Actual IPv4 settings used by the end-point. */
                IPV4Parameters_t ipv4_defaults; /**< Use values form "ipv4_defaults" in case DHCP has failed. */
            };
            #if ( ipconfigUSE_IPv6 != 0 )
                struct
                {
                    IPV6Parameters_t ipv6_settings; /**< Actual IPv6 settings used by the end-point. */
                    IPV6Parameters_t ipv6_defaults; /**< Use values form "ipv6_defaults" in case DHCP has failed. */
                };
            #endif
        } u;
```

as the endpoint structures are passed into the user-defined hooks [[vApplicationIPNetworkEventHook_Multi](https://www.freertos.org/FreeRTOS-Plus/FreeRTOS_Plus_TCP/API/vApplicationIPNetworkEventHook_Multi.html), [xApplicationDHCPHook_Multi](https://www.freertos.org/FreeRTOS-Plus/FreeRTOS_Plus_TCP/API/xApplicationDHCPHook_Multi.html), [xApplicationDNSQueryHook_Multi](https://www.freertos.org/FreeRTOS-Plus/FreeRTOS_Plus_TCP/API/xApplicationDNSQueryHook_Multi.html)], existing projects might already be using the ipv4 and ipv6 settings directly from the endpoint structure.

**Comment by tony-josi-aws** [comment]
@HTRamsey 
PR #1091 removes the unnamed union and structs from the endpoint structure so that it does not break any existing projects. Could you help in updating this PR once the changes in #1091 are taken in.

**Comment by HTRamsey** [comment]
@tony-josi-aws Yeah this doesn't entirely solve the issue either as there are still unnamed structs after the union. Perhaps I can remove those changes for now.

File: test/unit-test/FreeRTOS_DNS_Cache/FreeRTOS_DNS_Cache_utest.c [hunk 1]
```diff
@@ -630,7 +632,7 @@ void test_prepare_DNSLookup_IPv6( void )
     BaseType_t x = 0U;
     BaseType_t xFamily;
     struct freertos_addrinfo * pxAddressInfo = &pucAddrBuffer[ 0 ];
-    struct freertos_addrinfo ** ppxAddressInfo = &pucAddrBuffer[ 1 ];
+    struct freertos_addrinfo ** ppxAddressInfo = ( struct freertos_addrinfo ** ) &pucAddrBuffer;
     IPv46_Address_t xAddress;
 
     xFamily = FREERTOS_AF_INET6;
```
Comment (tony-josi-aws) [inline test/unit-test/FreeRTOS_DNS_Cache/FreeRTOS_DNS_Cache_utest.c:638]: [line: 643:](https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/pull/1090/files#diff-aaa8da95d9503fb80decbecf87cb0ed4d35cb3c1242e3e829c4e7130c928c3adR643) 
`struct freertos_addrinfo * pxAddrInfo = &pucAddrBuffer[ 2 ];`  

is accessing an out of bound array index. Could you help in updating that?

**Review by tony-josi-aws — DISMISSED** [review]

**Review by tony-josi-aws — APPROVED** [review]

**Review by shubnil — APPROVED** [review]

**Review by amazonKamath — APPROVED** [review]

**Pull request merged**